### ISSUE TITLE
No results for Kmer content module in HTML and fastqc_data.txt

### ISSUE THREAD

**Issue by RoadRoller** [guilhermesena1/falco#15]
Falco version is 0.2.4. Kmer module turned on in configuration by changing line in limits.txt to: 
```
kmer 				ignore 		0
```
Falco indicates that Kmer content FAIL, but there are no Kmer content in fastqc_data.txt:
```
>>Kmer Content	fail
#Sequence	Count	PValue	Obs/Exp Max	Max Obs/Exp Position
>>END_MODULE
```
In HTML file section not reported also:
![image](https://user-images.githubusercontent.com/740624/131095026-66cd24c8-238d-464f-a8a1-e2af80841d82.png)

For the same fastq file FastQC display Kmer content module as expected.
![image](https://user-images.githubusercontent.com/740624/131095297-ad8e481a-f823-404c-99ed-410159a35fec.png)

**Comment by guilhermesena1** [comment]
Hello,

Would you be able to help me reproduce the error on my end? Specifically, could you confirm

(1) if you cloned, used the release or installed from Conda
(2) could you tell me the exact command you used to run falco, and what it printed in the output and 
(3) if at all possible, could you share the first 40,000 lines of your input FASTQ file?

thank you!

**Comment by RoadRoller** [comment]
Hello,
thank you for response!
(1) I used the release and install it according to README.md with commands
$ ./configure CXXFLAGS="-O3 -Wall"
$ make all
$ make install
(2) exact command is
$ falco -l Configuration/limits.txt test.fastq
(before run command I turn on kmer module in limits.txt)
(3) this is config and head of fastq
[limits.txt](https://github.com/smithlabcode/falco/files/7109160/limits.txt)
[test.fastq.gz](https://github.com/smithlabcode/falco/files/7109163/test.fastq.gz)

**Comment by guilhermesena1** [comment]
So essentially the k-mer content functionality wasn't implemented at all! The data was being collected but not summarized properly. I went through the whole code and tested in your dataset. The fix on e115b55 should at least report and plot the k-mer contents, but it's not fully exactly like fastqc yet, specifically, I haven't implemented p-value calculations yet because I'll have to dig through the java BinomialDistribution package to know how to do it and it will take a little longer.

That said, I think log(p-values) are a monotonic function of the obs/exp ratio. In the current implementation I am sorting k-mers by obs/exp ratio and reporting the top ones (top 20 on fastqc_data, top 10 on the plots), which in most practical cases should be identical to the ones with lowest p-values.

If you use this at all in your data I'll be happy to know if this is working.

**Comment by RoadRoller** [comment]
Thank you! We will try 0.3.0 release.
